data-prepper-plugins/parquet-codecs/src/main/java/org/opensearch/dataprepper/plugins/codec/parquet/ParquetInputCodec.java

```diff
@@ -0,0 +1,111 @@
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.codec.InputCodec;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.log.JacksonLog;
+import org.opensearch.dataprepper.model.record.Record;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * An implementation of {@link InputCodec} which parses parquet records into fields.
+ */
+@DataPrepperPlugin(name = "parquet", pluginType = InputCodec.class)
+public class ParquetInputCodec implements InputCodec {
+
+    private static final String MESSAGE_FIELD_NAME = "message";
+    private static final String FILE_NAME = "parquet-data";
+    private static final String FILE_SUFFIX = ".parquet";
+
+    private static int fieldIndex;
+    private static final Logger LOG = LoggerFactory.getLogger(ParquetInputCodec.class);
+
+    @Override
+    public void parse(final InputStream inputStream, final Consumer<Record<Event>> eventConsumer) throws IOException {
+
+        Objects.requireNonNull(inputStream);
+        Objects.requireNonNull(eventConsumer);
+        parseParquetStream(inputStream, eventConsumer);
+
+    }
+
+    private void parseParquetStream(final InputStream inputStream, final Consumer<Record<Event>> eventConsumer) throws IOException {
+
+         final File tempFile = File.createTempFile(FILE_NAME, FILE_SUFFIX);
+        Files.copy(inputStream, tempFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+
+        try (ParquetFileReader parquetFileReader = new ParquetFileReader(HadoopInputFile.fromPath(new Path(tempFile.toURI()), new Configuration()), ParquetReadOptions.builder().build())) {
+            final ParquetMetadata footer = parquetFileReader.getFooter();
+            final MessageType schema = createdParquetSchema(footer);
+
+            PageReadStore pages;
+
+            while ((pages = parquetFileReader.readNextRowGroup()) != null) {
+                final long rows = pages.getRowCount();
+                final MessageColumnIO columnIO = new ColumnIOFactory().getColumnIO(schema);
+                final RecordReader recordReader = columnIO.getRecordReader(pages, new GroupRecordConverter(schema));
+
+                final Map<String, String> eventData = new HashMap<>();
+
+                for (int row = 0; row < rows; row++) {
+
+                        fieldIndex = 0;
+                        final SimpleGroup simpleGroup = (SimpleGroup) recordReader.read();
+                        for (Type field : schema.getFields()) {
+
+                            try {
+                                eventData.put(field.getName(), simpleGroup.getValueToString(fieldIndex, 0));
```

Comment:
This call to `getValueToString` is returning a `String`. However, the model may have some fields as integers, longs, or doubles.

Also, it appears that there may be sub-objects as indicated by the `getGroup` method.

Also, I noticed you are using `SimpleGroup`. This is in the package `org.apache.parquet.example.data.simple`. This makes me think it is not the right class to use.

https://www.javadoc.io/doc/org.apache.parquet/parquet-column/1.12.0/org/apache/parquet/example/data/simple/SimpleGroup.html